crates/subspace-networking/src/protocols/connected_peers/tests.rs

```diff
@@ -238,12 +243,13 @@ async fn test_target_connected_peer_limit_number() {
             _ = peer1.next_swarm_event().fuse() => {},
             _ = peer2.next_swarm_event().fuse() => {},
             _ = peer3.next_swarm_event().fuse() => {},
-            _ = sleep(large_delay).fuse() => {
+            _ = sleep(long_delay).fuse() => {
                 break;
             }
         }
     }
 
+    // We don't maintain with new peers when we have enough connected peers
```

Comment:
This is helpful, but there are 3 blocks. The comment explains the high idea behind the whole test, it doesn't really comment what is happening here. And what is happening in these 3 blocks of assertions is checks where some peers are within limits and thus maintain connections and others are not and thus don't maintain connections.